src/components/Stopwatch/Stopwatch.js

```diff
@@ -0,0 +1,124 @@
+import React from "react";
+import "./Stopwatch.css";
+
+export default class Stopwatch extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hours: 0,
+      minutes: 0,
+      seconds: 0,
+      intervalID: "",
+      startVisible: true,
+      // TODO: remove stopVisible
+      stopVisible: false,
+      lapTimes: []
+    };
+
+    this.handleStart = this.handleStart.bind(this);
+    this.handleReset = this.handleReset.bind(this);
+    this.handleStop = this.handleStop.bind(this);
+    this.handleAddLap = this.handleAddLap.bind(this);
+  }
+
+  handleStart() {
+    this.setState({
+      startVisible: false,
+      stopVisible: true
+    });
+    let interval = setInterval(() => {
+      if (this.state.minutes === 59) {
+        this.setState({
+          hours: this.state.hours + 1,
+          minutes: 0
+        });
+      }
+      if (this.state.seconds === 59) {
+        this.setState({
+          minutes: this.state.minutes + 1,
+          seconds: 0
+        });
+      } else {
+        this.setState({
+          seconds: this.state.seconds + 1
+        });
+      }
+    }, 1000);
+
+    this.setState({ intervalID: interval });
+  }
+
+  handleStop() {
+    clearInterval(this.state.intervalID);
+    this.setState({
+      startVisible: true,
+      stopVisible: false
+    });
+  }
+
+  handleReset() {
+    this.setState({
+      seconds: 0,
+      minutes: 0,
+      hours: 0
+    });
+  }
+
+  handleAddLap() {
+    console.log(this.state);
+    this.setState(state => {
+      const lapTimes = [
+        ...state.lapTimes,
+        { hours: state.hours, minutes: state.minutes, seconds: state.seconds }
+      ];
+      return { lapTimes, hours: 0, minutes: 0, seconds: 0 };
+    });
+  }
+
+  toDoubleDigit(digit) {
+    return digit < 10 ? "0" + digit : digit;
```

Comment:
Personally I'd prefer either `` `0${digit}` `` or `String(digit).padStart(2, '0')`. 

That `string + number` concatenates to a string is kind of a quirk of JS (lots of other languages will throw errors), and I don't like having these kinds of surprises in my code because it causes headscratching for readers who come from other language backgrounds.

Reply:
Nice, I didn't know about that first notation `0${digit}`. I'll see where I can use that from now on.